hi-ml-histopathology/src/histopathology/models/transforms.py

```diff
@@ -43,6 +43,37 @@ def load_image_stack_as_tensor(image_paths: Sequence[PathOrString],
     return torch.stack(image_tensors, dim=0)
 
 
+def transform_dict_adaptor(function: Callable, k_input: str = None, k_output: str = None) -> Callable:
+    """Adapt transformations to work with an input dictionary (rather than a tensor).
+       We can't reuse monai.transforms.adaptors because it is only compatible with transformations that accept
+       a dict as input.
+
+    :param function: a transformation function
+    :param k_input: key of the input dictionary that contains the object
+        to which function should be applied
+    :param k_output: key of the input dictionary where to place the function output. If None the ouput of
+        the transformation is returned
+
+    :return: adapted transformation
+    """
+    def _inner(ditems: dict) -> Dict:
+        if k_input is None:
+            dinputs = ditems
+        else:
+            dinputs = ditems[k_input]
+        ret = function(dinputs)
+        if k_output is None:
+            ditems = ret
+        else:
+            if isinstance(ret, type(ditems[k_output])):
```

Comment:
For my information, I'm not sure I understand this. Is the dictionary expected to already contain `k_output`? And why do the types need to match? This seems overly restrictive—consider e.g. loading an image tensor from a path.

Another comment is that the MONAI docs suggest that transforms shouldn't modify input dictionaries in-place, but rather call `ditems.copy()` before assigning the outputs.

Reply:
I think they're both good comments:
- correct we could extend this adaptor to be compatible with a broader set of transformations and also give the option to add a key rather than just overwriting. I wanted to keep this as simple as possible to solve our current case.
- do you know why MONAI suggest to copy? here I am a bit worried about memory consumption if every transformation is creating a copy but happy to follow up on this

Reply:
On the second point, it's just a shallow copy of the dictionary, in case the original needs to be used elsewhere (e.g. if passed to multiple parallel transformations).